src/sshfs_mount/sftp_server.h

```diff
@@ -76,13 +78,16 @@ class SftpServer
     int handle_write(sftp_client_message msg);
     int handle_extended(sftp_client_message msg);
 
+    template <typename T>
+    T* get_handle(sftp_client_message msg);
+
     SSHSession ssh_session;
     SSHFSProcUptr sshfs_process;
     SftpSessionUptr sftp_server_session;
     const std::string source_path;
     const std::string target_path;
-    std::unordered_map<void*, std::unique_ptr<QFileInfoList>> open_dir_handles;
-    std::unordered_map<void*, std::unique_ptr<QFile>> open_file_handles;
+    std::unordered_map<void*, std::unique_ptr<NamedFd, void (*)(NamedFd*)>> open_file_handles;
```

Comment:
Maybe change `void (*)(NamedFd*)` from `std::function<void(NamedFd*)`, so we have a more readable functor signature.

Reply:
Good call. At first I avoided `std::function` because I thought it always involves a dynamic allocation, but after reading the docs better, I understand that in the case of free function pointers, small object optimization is guaranteed and no dynamic allocation takes place. So I'll change it.